tests/transforms/test_optimization/test_optimization_utils.py

```diff
@@ -107,6 +107,8 @@ def test_quaternion_product(self, angles_1, angles_2, expected_quat):
             ([0.05, 0.2, 0.0], [0.0, -0.6, 0.12]),
             ([0.05, -1.34, 4.12], [0.0, 0.2, 0.12]),
             ([0.05, -1.34, 4.12], [0.3, 0.0, 0.12]),
+            ([np.pi, np.pi / 2, 0.0], [0.0, -np.pi / 2, 0.0]),
+            ([0.9, np.pi / 2, 0.0], [0.0, -np.pi / 2, 0.0]),
```

Comment:
These seem to be special cases when RY is cancelling out. Could we have such examples for RX and RZ too to be on the safe side?

Reply:
Probably this is not necessary, because the singular point is at exactly this rotation angles. However, it doesn't cost much and to be on the safe side sounds good :)

Reply:
Done :)